File: 2966-divide-array-into-arrays-with-max-difference/2966-divide-array-into-arrays-with-max-difference.py
class Solution:
    def divideArray(self, nums: List[int], k: int) -> List[List[int]]:
        n = len(nums)
        nums.sort()
        for i in range(2, n, 3):
            if nums[i] - nums[i - 2] > k:
                return []

        result = [nums[i:i + 3] for i in range(0, n, 3)]
        return result

File: 2966-divide-array-into-arrays-with-max-difference/2966-divide-array-into-arrays-with-max-difference.java
class Solution {
    public int[][] divideArray(int[] nums, int k) {
        int n=nums.length;
        int[][] arr=new int[n/3][3];
        Arrays.sort(nums);
        for(int i=0;i<n;i+=3){
            if(nums[i+2]-nums[i]>k) return new int[0][0];
            int[] temp=new int[]{nums[i],nums[i+1],nums[i+2]};
            arr[i/3]=temp;
        }
        return arr;
    }
}